admintools/middleware.py

```diff
@@ -23,6 +13,12 @@ def process_request(self, request):
                     emulated_user = get_user_by_id(user_id)
                 elif username:
                     emulated_user = get_user_by_username(username)
-                if emulated_user:
+                if can_emulate_user(request.user, emulated_user):
                     request.original_user = request.user
                     request.user = emulated_user
+
+    def process_response(self, request, response):
+        if can_emulate_another_user(request.user):
```

Comment:
This logic is fine. If I change it to `if not` it will delete cookies every time we have a successful emulation.

Consider the three possible users.
Bad User (non-existent user): Our request.user will be the original user since we never switched. So if this user happens to be a super user, then we delete the cookie. If not, then this other case doesn't ever happen because it will be the `good user` case
Good User: The request.user will now be our targeted user. This user cannot emulate, so we don't want to delete cookies, let things pass as normal
Employee User: We are able to successfully get this user, but this user is a super user. So we can emulate, therefore we want to remove the cookie.

Reply:
@ajnassar you are missing my point, entirely

can you do something like

```
if not hasattr(request, 'original_user') or not can_emulate_another_user(request.user):
```